### DISCUSSION TITLE
Is there a way to specify hide duration on NavigationMenu?

### DISCUSSION THREAD

**Discussion by nikosrossolatos** [radix-ui/primitives#1999] [Help] [answered]
Hi!

Is there a way to hide immediately/with-custom-delay the navigation menu instead of having a fixed 200ms delay? Something like what the delayDuration does, but for when we hide the component!

Thanks

**Comment by joaom00** [comment, marked as answer]
If you want to hide immediately you can remove the animation when closed but if you want something like delayDuration for close there isn't

**Reply by nikosrossolatos** [reply]
It seems like there's a 200ms delay before closing the menu that it seems like it's not an animation?
Here's a stripped down sandbox of the navigation menu
https://codesandbox.io/p/sandbox/stupefied-johnson-wrxqet

**Reply by joaom00** [reply]
Yeah, you are right there's a delay. I don't know how to remove it. Maybe @andy-hook can guide us here :pray:

**Reply by benoitgrelard** [reply]
@andy-hook do you remember why the 150ms is hardcoded here: https://github.com/radix-ui/primitives/pull/1716/files#diff-8e96b8f4b466ef7144a8f478a3e0c8afcecc7f42ab59b0af0595889df64f1c28R145 ?

**Reply by andy-hook** [reply]
This is a grace period so that the pointer can transition from the `Content` to the `Trigger`. Ideally this would use the same [solution as the tooltip](https://github.com/radix-ui/primitives/pull/1490), avoiding the need for a delay entirely, we just haven't found time to implement this yet.

A solution for now would be to disable the leave events [per this sandbox](https://codesandbox.io/s/navigation-menu-on-click-duwvgn?file=/App.js:8562-8625) and use the controlled api combined with your own exit events.

**Comment by mwkcoding** [comment]
Is there still no proper solution to this?
